be/src/vec/sink/vtablet_sink.cpp

```diff
@@ -426,6 +434,15 @@ Status VOlapTableSink::send(RuntimeState* state, vectorized::Block* input_block)
     if (findTabletMode == FindTabletMode::FIND_TABLET_EVERY_BATCH) {
         _partition_to_tablet_map.clear();
     }
+
+    //if pending bytes is more than 500M, wait
+    constexpr size_t MAX_PENDING_BYTES = 500 * 1024 * 1024;
+    if (get_pending_bytes() > MAX_PENDING_BYTES) {
+        while (get_pending_bytes() < MAX_PENDING_BYTES) {
+            std::this_thread::sleep_for(std::chrono::microseconds(500));
```

Comment:
Is it necessary to set a maximum waiting time?

Reply:
I will remove these code. And add more check logic in next pr. This pr is focused cherry-pick from 1.0.1